cpp/src/arrow/util/parsing.h

```diff
@@ -444,6 +468,39 @@ class StringConverter<TimestampType> {
     return out->ok();
   }
 
+  bool ParseHH(const char* s, std::chrono::duration<value_type>* out) {
```

Comment:
Should we reuse `ParseHH_MM_SS()` instead of defining similar parse functions?

Reply:
This solution seems pretty straightforward and perhaps more performant.